lang/java/reef-common/src/main/java/org/apache/reef/io/WorkingDirectoryTempFileCreator.java

```diff
@@ -40,7 +40,7 @@
 
   @Inject
   WorkingDirectoryTempFileCreator() throws IOException {
-    this.tempFolderAsFile = new File("./reef/temp");
+    this.tempFolderAsFile = new File("./target/reef/temp");
```

Comment:
@markusweimer This change will affect temporary directories created in YARN, Mesos, and HDInsight as well as local test runs. Is this the behavior we want?